streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -490,7 +487,7 @@ private void replaceStreamThread(final Throwable throwable) {
             closeToError();
         }
         final StreamThread deadThread = (StreamThread) Thread.currentThread();
-        deadThread.shutdown();
+        deadThread.shutdown(true);
```

Comment:
It seems the origin behavior does not want to leave group, right? otherwise, it will trigger the extra rebalance.